In [82]:
#Import all important libraries
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline
import warnings
warnings.filterwarnings('ignore')

In [83]:
data = pd.read_excel('datasets/Data_Train.xlsx')

In [84]:
data.head(2)

Airline Date_of_Journey    Source Destination                  Route  \
0     IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1  Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   

  Dep_Time  Arrival_Time Duration Total_Stops Additional_Info  Price  
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897  
1    05:50         13:15   7h 25m     2 stops         No info   7662

In [85]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10683 entries, 0 to 10682
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Airline          10683 non-null  object
 1   Date_of_Journey  10683 non-null  object
 2   Source           10683 non-null  object
 3   Destination      10683 non-null  object
 4   Route            10682 non-null  object
 5   Dep_Time         10683 non-null  object
 6   Arrival_Time     10683 non-null  object
 7   Duration         10683 non-null  object
 8   Total_Stops      10682 non-null  object
 9   Additional_Info  10683 non-null  object
 10  Price            10683 non-null  int64 
dtypes: int64(1), object(10)
memory usage: 918.2+ KB


In [86]:
data.describe()

Price
count  10683.000000
mean    9087.064121
std     4611.359167
min     1759.000000
25%     5277.000000
50%     8372.000000
75%    12373.000000
max    79512.000000

In [87]:
data.shape

(10683, 11)

In [88]:
highest_price= data[data['Price'] == data['Price'].max()]

In [89]:
highest_price

Airline Date_of_Journey    Source Destination  \
2924  Jet Airways Business      01/03/2019  Banglore   New Delhi   

                Route Dep_Time Arrival_Time Duration Total_Stops  \
2924  BLR → BOM → DEL    05:45        11:25   5h 40m      1 stop   

     Additional_Info  Price  
2924  Business class  79512

In [90]:
lowest_price= data[data['Price'] == data['Price'].min()]

In [91]:
lowest_price

Airline Date_of_Journey  Source Destination      Route Dep_Time  \
4066   SpiceJet      21/03/2019  Mumbai   Hyderabad  BOM → HYD    22:45   
4274   SpiceJet      27/03/2019  Mumbai   Hyderabad  BOM → HYD    22:45   
4839   SpiceJet       3/04/2019  Mumbai   Hyderabad  BOM → HYD    22:45   
10513  SpiceJet      27/03/2019  Mumbai   Hyderabad  BOM → HYD    05:45   

       Arrival_Time Duration Total_Stops Additional_Info  Price  
4066   00:10 22 Mar   1h 25m    non-stop         No info   1759  
4274   00:10 28 Mar   1h 25m    non-stop         No info   1759  
4839   00:15 02 Apr   1h 30m    non-stop         No info   1759  
10513         07:05   1h 20m    non-stop         No info   1759

In [92]:
data.isnull().sum()

Airline            0
Date_of_Journey    0
Source             0
Destination        0
Route              1
Dep_Time           0
Arrival_Time       0
Duration           0
Total_Stops        1
Additional_Info    0
Price              0
dtype: int64

In [93]:
data[data['Route'].isnull() == True]

Airline Date_of_Journey Source Destination Route Dep_Time  \
9039  Air India       6/05/2019  Delhi      Cochin   NaN    09:45   

      Arrival_Time Duration Total_Stops Additional_Info  Price  
9039  09:25 07 May  23h 40m         NaN         No info   7480

In [94]:
data["Date"]=data['Date_of_Journey'].str.split('/').str[0]
data["Month"]=data['Date_of_Journey'].str.split('/').str[1]
data["Year"]=data['Date_of_Journey'].str.split('/').str[2]

In [95]:
data["Date"]=data["Date"].astype('int')
data["Month"]=data["Month"].astype('int')
data["Year"]=data["Year"].astype('int')

In [96]:
data.drop("Date_of_Journey",inplace=True,axis=1)

In [97]:

import numpy as np

In [98]:
np.quantile(data['Date'],[0.0,0.25,0.50,0.75,1.0])

array([ 1.,  6., 12., 21., 27.])

In [99]:
data["Date"].value_counts()

Date
9     1406
6     1288
27    1130
21    1111
1     1075
24    1052
15     984
12     957
3      848
18     832
Name: count, dtype: int64

In [100]:
data['Airline'].value_counts()

Airline
Jet Airways                          3849
IndiGo                               2053
Air India                            1752
Multiple carriers                    1196
SpiceJet                              818
Vistara                               479
Air Asia                              319
GoAir                                 194
Multiple carriers Premium economy      13
Jet Airways Business                    6
Vistara Premium economy                 3
Trujet                                  1
Name: count, dtype: int64

## Insight
##### highest ticket price was in 2019  Jet Airways Business price 79512 
##### lowest ticket price was in 2019  SpiceJet price 1759
##### Maximum people travells in between date of 6,9,21 and 27 of every month
##### Maximum people travels in Jet Ariways ,IndiGo  and Air India.

In [101]:
data.head(2)

Airline    Source Destination                  Route Dep_Time  \
0     IndiGo  Banglore   New Delhi              BLR → DEL    22:20   
1  Air India   Kolkata    Banglore  CCU → IXR → BBI → BLR    05:50   

   Arrival_Time Duration Total_Stops Additional_Info  Price  Date  Month  Year  
0  01:10 22 Mar   2h 50m    non-stop         No info   3897    24      3  2019  
1         13:15   7h 25m     2 stops         No info   7662     1      5  2019

In [102]:
data["T_Hour"]=data['Duration'].str.split(' ').str[0].str.split('h').str[0]
data["T_Minute"]=data['Duration'].str.split(' ').str[1].str.split('m').str[0]

In [103]:
data["T_Minute"] = data["T_Minute"].astype(int)

ValueError: cannot convert float NaN to integer

In [ ]:
data["T_Minute"].shape

In [ ]:
data[data["T_Minute"].str.isnumeric()]

In [ ]:
data[data["T_Minute"].str.contains('m')]

In [ ]:
data["T_Hour"]=data["T_Hour"].astype(int)
data["T_Minute"]=data["T_Minute"].astype(int)

In [ ]:
data.drop('Duration',inplace=True,axis=1)

In [ ]:
data.head(2)

In [ ]:
# extracting hour and minutes from arrival time to get numerical data in new columns
data["Arr_Hour"]=data['Arrival_Time'].str.split(' ').str[0].str.split(':').str[0]
data["Arr_Minute"]=data['Arrival_Time'].str.split(' ').str[0].str.split(':').str[1]
# extracting hour and minutes from departure time to get numerical data in new columns
data["Dep_Hour"]=data['Dep_Time'].str.split(':').str[0]
data["Dep_Minute"]=data['Dep_Time'].str.split(':').str[1]

In [ ]:
#droping arrival time columns
data.drop('Arrival_Time',inplace = True,axis=1)
#droping departure time columns
data.drop('Dep_Time',inplace = True,axis=1)

In [ ]:
data["Arr_Hour"]=data["Arr_Hour"].astype(int)
data["Arr_Minute"]=data["Arr_Minute"].astype(int)
data["Dep_Hour"]=data["Dep_Hour"].astype(int)
data["Dep_Minute"]=data["Dep_Minute"].astype(int)

In [ ]:
data.head(5)

In [ ]:
data["T_Minute"].mean()

In [ ]:
data.shape[0]